src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -2131,6 +2134,65 @@ private void checkInternetPermission() {
         }
     }
 
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if ((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            /* the current serialize/deserialize feature is for first party 
+             * so the client ID for the FoCI token cache item is hard coded below
+             */
```

Comment:
@weijjia In the FoCI implementation do you define the term FoCI anywhere.  I think that we need to add a comment with a definition somewhere so that this term doesn't come out of nowhere.